go_modules/spec/dependabot/go_modules/update_checker/latest_version_finder_spec.rb

```diff
@@ -238,4 +241,56 @@ module foobar
       end
     end
   end
+
+  describe "#lowest_security_fix_version" do
+    subject { finder.lowest_security_fix_version }
+
+    let(:current_version) { "1.0.0" }
+
+    context "when on a stable release and a newer versions are available" do
+      it "returns the lowest available new release" do
+        expect(finder.lowest_security_fix_version).to eq(Dependabot::GoModules::Version.new("1.0.1"))
+      end
+    end
+
+    context "with Dependabot-ignored versions" do
+      let(:ignored_versions) { ["= 1.0.1"] }
+
+      it "doesn't return Dependabot-ignored versions" do
+        expect(finder.lowest_security_fix_version).to eq(Dependabot::GoModules::Version.new("1.0.5"))
+      end
+    end
+
+    context "with a go.mod vulnerable version" do
+      let(:security_advisories) do
+        [
+          Dependabot::SecurityAdvisory.new(
+            dependency_name: dependency_name,
+            package_manager: "go_modules",
+            vulnerable_versions: ["<= 1.0.5"]
+          )
+        ]
+      end
+
+      it "doesn't return to the vulnerable version" do
+        expect(finder.lowest_security_fix_version).to eq(Dependabot::GoModules::Version.new("1.0.6"))
+      end
+    end
+
+    context "when on a pre-release" do
+      let(:dependency_version) { "1.2.0-pre1" }
+
+      it "returns newest pre-release" do
+        expect(finder.lowest_security_fix_version).to eq(Dependabot::GoModules::Version.new("1.2.0-pre2"))
+      end
+    end
+
+    context "when on a stable release and a newer prerelease is available" do
+      let(:current_version) { "1.1.0" }
+
+      it "doesn't return pre-release" do
+        expect(finder.lowest_security_fix_version).to_not eq(Dependabot::GoModules::Version.new("1.2.0-pre2"))
```

Comment:
Could this test the positive case instead and expect it to `eq` a version? Can be hard to catch regressions when only checking the negative case, as a new pre-release version could still make this pass